### PR TITLE
Remove ts-ignore for initMergeProps

### PR DESCRIPTION
**Pull request by Methuselah96**

### CODE REVIEW

File: src/components/connect.tsx [hunk 1]
```diff
@@ -541,7 +504,6 @@ function connect<
       initMapStateToProps,
       // @ts-ignore
       initMapDispatchToProps,
-      // @ts-ignore
       initMergeProps,
```
Comment (Methuselah96) [inline src/components/connect.tsx:507]: The goal of this PR was to remove this `@ts-ignore`.

File: src/components/connect.tsx [hunk 2]
```diff
@@ -485,24 +463,9 @@ function connect<
...
-    // @ts-ignore
-    defaultMapDispatchToPropsFactories,
-    'mapDispatchToProps'
-  )!
-  const initMergeProps = match(
-    mergeProps,
-    // @ts-ignore
-    defaultMergePropsFactories,
-    'mergeProps'
-  )!
```
Comment (Methuselah96) [inline src/components/connect.tsx:505]: The `match` function is not TypeScript friendly. I refactored those to just be functions (that I think are just as easy to understand as before).

File: src/connect/mapDispatchToProps.ts [hunk 3]
```diff
@@ -1,36 +1,25 @@
...
-  mapDispatchToProps: ActionCreatorsMapObject
+export function mapDispatchToPropsFactory<TDispatchProps, TOwnProps>(
+  mapDispatchToProps:
+    | MapDispatchToPropsParam<TDispatchProps, TOwnProps>
+    | undefined
 ) {
   return mapDispatchToProps && typeof mapDispatchToProps === 'object'
-    ? wrapMapToPropsConstant((dispatch: Dispatch) =>
+    ? wrapMapToPropsConstant((dispatch: Dispatch<Action<unknown>>) =>
+        // @ts-ignore
```
Comment (Methuselah96) [inline src/connect/mapDispatchToProps.ts:14]: To be fixed in a later PR.

File: src/connect/mapDispatchToProps.ts [hunk 4]
```diff
@@ -1,36 +1,25 @@
...
         bindActionCreators(mapDispatchToProps, dispatch)
```
Comment (Methuselah96) [inline src/connect/mapDispatchToProps.ts:15]: To be fixed in a later PR.

File: src/connect/mapDispatchToProps.ts [hunk 5]
```diff
@@ -1,36 +1,25 @@
...
       )
-    : undefined
+    : !mapDispatchToProps
+    ? wrapMapToPropsConstant((dispatch: Dispatch<Action<unknown>>) => ({
+        dispatch,
+      }))
+    : typeof mapDispatchToProps === 'function'
+    ? // @ts-ignore
+      wrapMapToPropsFunc(mapDispatchToProps, 'mapDispatchToProps')
```
Comment (Methuselah96) [inline src/connect/mapDispatchToProps.ts:23]: To be fixed in a later PR.

File: src/connect/mapStateToProps.ts [hunk 6]
```diff
@@ -1,17 +1,14 @@
...
-    ? wrapMapToPropsFunc(mapStateToProps, 'mapStateToProps')
-    : undefined
+export function mapStateToPropsFactory<TStateProps, TOwnProps, State>(
+  mapStateToProps: MapStateToPropsParam<TStateProps, TOwnProps, State>
+) {
+  return !mapStateToProps
+    ? wrapMapToPropsConstant(() => ({}))
+    : typeof mapStateToProps === 'function'
+    ? // @ts-ignore
+      wrapMapToPropsFunc(mapStateToProps, 'mapStateToProps')
```
Comment (Methuselah96) [inline src/connect/mapStateToProps.ts:12]: To be fixed in a later PR.

**Comment by markerikson** [comment]
Nice!

**Pull request merged**